src/main/java/io/reactivex/Flowable.java

```diff
@@ -10650,7 +10650,7 @@ public final Completable ignoreElements() {
     @CheckReturnValue
     @BackpressureSupport(BackpressureKind.UNBOUNDED_IN)
     @SchedulerSupport(SchedulerSupport.NONE)
-    public final Maybe<T> reduce(BiFunction<T, T, T> reducer) {
+    public final Maybe<T> reduce(BiFunction<? super T, ? super  T, ? extends T> reducer) {
```

Comment:
I don't want to keep pushing a finished topic. But just for your interest: This code compiles (have tried with 1.6 and 1.8):

I have added this test case to this pull request, if anybody wants to take a look.

````
@Test
    public void basics() throws Exception {
        Flowable<Integer> flowable = Flowable.just(1, 2, 3, 4);

        BiFunction<Integer, Integer, Integer> sumReducer = new BiFunction<Integer, Integer, Integer>() {
            @Override
            public Integer apply(Integer integer, Integer integer2) throws Exception {
                return integer + integer2;
            }
        };

        Maybe<Integer> reduced = flowable.reduce(sumReducer);

        reduced.subscribe(new Consumer<Integer>() {
            @Override
            public void accept(Integer integer) throws Exception {
                assertEquals(10, integer, 0);
            }
        });
    }

    @Test
    public void reduceWithSuperType() throws Exception {
        Flowable<Integer> flowable = Flowable.just(1, 2, 3, 4);

        BiFunction<Number, Number, Integer> sumReducer = new BiFunction<Number, Number, Integer>() {
            @Override
            public Integer apply(Number integer, Number integer2) throws Exception {
                int value = integer.intValue() + integer2.intValue();
                return value;
            }
        };

        Maybe<Integer> reduced = flowable.reduce(sumReducer);
        reduced.subscribe(new Consumer<Integer>() {
            @Override
            public void accept(Integer integer) throws Exception {
                assertEquals(10, integer, 0);
            }
        });
    }

    @Test
    public void reduceWithSuperTypeNumberFlowable() throws Exception {
        Flowable<Number> flowable = Flowable.<Number>just(1, 2, 3, 4);

        BiFunction<Number, Number, Integer> sumReducer = new BiFunction<Number, Number, Integer>() {
            @Override
            public Integer apply(Number integer, Number integer2) throws Exception {
                int value = integer.intValue() + integer2.intValue();
                return value;
            }
        };

        Maybe<Number> reduced = flowable.reduce(sumReducer);
        reduced.subscribe(new Consumer<Number>() {
            @Override
            public void accept(Number integer) throws Exception {
                assertEquals(10, integer.doubleValue(), 0);
            }
        });
    }

    @Test
    public void reduceWithSuperTypeNumberFlowable2() throws Exception {
        Flowable<Number> flowable = Flowable.<Number>just(1, 2, 3, 4);

        BiFunction<Number, Number, Number> sumReducer = new BiFunction<Number, Number, Number>() {
            @Override
            public Number apply(Number integer, Number integer2) throws Exception {
                int value = integer.intValue() + integer2.intValue();
                return value;
            }
        };

        Maybe<Number> reduced = flowable.reduce(sumReducer);
        reduced.subscribe(new Consumer<Number>() {
            @Override
            public void accept(Number integer) throws Exception {
                assertEquals(10, integer.doubleValue(), 0);
            }
        });
    }